src/middlewared/middlewared/utils/os.py

```diff
@@ -51,3 +63,26 @@ def terminate_pid(pid: int, timeout: int = 10) -> bool:
     except ProcessLookupError:
         # Process may have terminated between checks
         return True
+
+
+def get_pids(pid: int | None = None) -> Generator[PidEntry] | PidEntry | None:
+    """Get the currently running processes on the OS.
+
+    pid: int if provided, will short-circuit and return a
```

Comment:
Can we just have `get_pids(pid)` and `get_pid(pid)`? This signature overload style is really confusing. And `Generator[PidEntry] | PidEntry | None` return type will help little for OS autocompletion. I also don't see `get_pids(pid)` being used anywhere.

Reply:
Heh, I actually wrote it originally like you suggested but tried to be clever. I'll go back to how I originally did it (what you suggest).